packages/sdk/src/evm/core/sdk.ts

```diff
@@ -309,6 +311,21 @@ export class ThirdwebSDK extends RPCConnectionHandler {
       this.options,
       this.storageHandler,
     );
+
+    if (
+      typeof globalThis !== "undefined" &&
+      (globalThis as any).X_SDK_NAME === undefined
+    ) {
+      (globalThis as any).X_SDK_NAME = pkg.name;
+      (globalThis as any).X_SDK_PLATFORM =
+        typeof navigator !== "undefined" && navigator.product === "ReactNative"
+          ? "mobile"
+          : typeof window !== "undefined"
+            ? "browser"
+            : "node";
+      (globalThis as any).X_SDK_VERSION = pkg.version;
+      (globalThis as any).X_SDK_OS = getOperatingSystem();
```

Comment:
hmmm i wonder if we should just do these as fallbacks at the callsite where we pass the headers instead. Bit worried about the order of operations here